generator/wrapper_gen.py

```diff
@@ -175,12 +175,13 @@ def my_procid():
 
 def _get_data(x):
     if isinstance(x, pdDataFrame):
-        if x._is_homogeneous_type:
-            x = x.values
+        x_dtypes = x.dtypes
+        if np.all(x_dtypes == x_dtypes[0]):
+            x = x.to_numpy()
```

Comment:
We need to check contiguity; homogeneity is only a required but not sufficient attribute.